src/services/api/handlers/packageActions.ts

```diff
@@ -12,36 +12,130 @@ const config = {
   database: "SEA",
 };
 
-import { Action, OneMacSink, transformOnemac } from "shared-types";
+import { Action, raiSchema, RaiSchema } from "shared-types";
 import { produceMessage } from "../libs/kafka";
 import { response } from "../libs/handler";
+import { SEATOOL_STATUS } from "shared-types/statusHelper";
+import { getActiveRai, getLatestRai } from "shared-utils";
 
 const TOPIC_NAME = process.env.topicName;
 
-export async function issueRai(id: string, timestamp: number) {
+export async function issueRai(body: RaiSchema) {
   console.log("CMS issuing a new RAI");
   const pool = await sql.connect(config);
-  const query = `
-    Insert into SEA.dbo.RAI (ID_Number, RAI_Requested_Date)
-      values ('${id}'
-        ,dateadd(s, convert(int, left(${timestamp}, 10)), cast('19700101' as datetime)))
-  `;
-  // Prepare the request
-  const request = pool.request();
-  request.input("ID_Number", sql.VarChar, id);
-  request.input("RAI_Requested_Date", sql.DateTime, new Date(timestamp));
-
-  const result = await sql.query(query);
-  console.log(result);
-  await pool.close();
+  const transaction = new sql.Transaction(pool);
+  try {
+    await transaction.begin();
+    // Issue RAI
+    const query1 = `
+      Insert into SEA.dbo.RAI (ID_Number, RAI_Requested_Date)
+        values ('${body.id}'
+        ,dateadd(s, convert(int, left(${body.requestedDate}, 10)), cast('19700101' as datetime)))
+    `;
+    const result1 = await transaction.request().query(query1);
+    console.log(result1);
+
+    // Update Status
+    const query2 = `
+      UPDATE SEA.dbo.State_Plan
+        SET SPW_Status_ID = (Select SPW_Status_ID from SEA.dbo.SPW_Status where SPW_Status_DESC = '${SEATOOL_STATUS.PENDING_RAI}')
+        WHERE ID_Number = '${body.id}'
+    `;
+    const result2 = await transaction.request().query(query2);
+    console.log(result2);
+
+    // write to kafka here
+    const result = raiSchema.safeParse(body);
+    if (result.success === false) {
+      console.log(
+        "RAI Validation Error. The following record failed to parse: ",
+        JSON.stringify(body),
+        "Because of the following Reason(s):",
+        result.error.message
+      );
+    } else {
+      await produceMessage(
+        TOPIC_NAME,
+        body.id,
+        JSON.stringify({ ...result.data, actionType: Action.ISSUE_RAI })
+      );
+    }
+
+    // Commit transaction
+    await transaction.commit();
+  } catch (err) {
+    // Rollback and log
+    await transaction.rollback();
+    console.error("Error executing one or both queries:", err);
```

Comment:
Because this catches the error at this scope and doesn't throw to the `try/catch` in `action.ts`, you'll still get a `200` response when this function completes, I think. 

I added the following in to help it, but we can also just `throw(err)` and the catch block in the endpoint's main handler will give the same 500 back:

```javascript
return response({
    statusCode: 500,
    body: { message: err.message },
});
```